File: cloud-functions/submitTrade/exchanges/bingx/scripts/orderFactory.js
class Order {
    constructor(symbol, type, side, price, quantity, positionSide, stopPrice, clientOrderID) {
        this.symbol = symbol;
        this.type = type;
        this.side = side;
        this.price = price
        this.quantity = quantity;
        this.positionSide = positionSide;
        this.stopPrice = stopPrice;
        this.clientOrderID = clientOrderID;
    }
}

module.exports = Order;

File: cloud-functions/submitTrade-py/exchanges/bingx/scripts/order_factory.py
class Order:
    def __init__(self, symbol, type, side, price, quantity, positionSide, stopPrice, clientOrderID):
        self.symbol = symbol
        self.type = type
        self.side = side
        self.price = price
        self.quantity = quantity
        self.positionSide = positionSide
        self.stopPrice = stopPrice
        self.clientOrderID = clientOrderID

    def remove_none_attributes(self):
        to_delete = [key for key, value in self.__dict__.items() if value is None]
        for key in to_delete:
            delattr(self, key)
